src/Microsoft.AspNet.OData.Shared/Query/HandleNullPropagationOptionHelper.cs

```diff
@@ -9,11 +9,11 @@ namespace Microsoft.AspNet.OData.Query
 {
     internal static class HandleNullPropagationOptionHelper
     {
-        private const string EntityFrameworkQueryProviderNamespace = "System.Data.Entity.Internal.Linq";
+        internal const string EntityFrameworkQueryProviderNamespace = "System.Data.Entity.Internal.Linq";
 
-        private const string ObjectContextQueryProviderNamespaceEF5 = "System.Data.Objects.ELinq";
-        private const string ObjectContextQueryProviderNamespaceEF6 = "System.Data.Entity.Core.Objects.ELinq";
-        private const string ObjectContextQueryProviderNamespaceEFCore2 = "Microsoft.EntityFrameworkCore.Query.Internal";
+        internal const string ObjectContextQueryProviderNamespaceEF5 = "System.Data.Objects.ELinq";
+        internal const string ObjectContextQueryProviderNamespaceEF6 = "System.Data.Entity.Core.Objects.ELinq";
+        internal const string ObjectContextQueryProviderNamespaceEFCore2 = "Microsoft.EntityFrameworkCore.Query.Internal";
 
         private const string Linq2SqlQueryProviderNamespace = "System.Data.Linq";
```

Comment:
>System.Data.Linq [](start = 63, length = 16)

Are we correctly handling Linq2SqlQueryProviderNamespace, or did we change defaults for this (and other) non-ef linq providers? Should non-ef linq providers have the _classicEF behavior? Or would the previous code (likely) not have worked for these providers?

Reply:
Previous version was never tested with LINQ 2 SQL. I don't see any LINQ2SQL tests in the repo. I could try to add them.

Reply:
I did some digging and Linq2SQL doesn't work with aggregations even without my changes. I managed to support only simple case like` $apply=aggregate($count as Count)`, but as soon as I'm adding groupby or more expressions to aggregate I'm getting exceptions like:

```
Unhandled Exception: System.NotSupportedException: The member 'GroupByWrapper.Container' has no supported translation to SQL.
   at System.Data.Linq.SqlClient.PostBindDotNetConverter.Visitor.VisitMember(SqlMember m)
   at System.Data.Linq.SqlClient.SqlVisitor.Visit(SqlNode node)
   at System.Data.Linq.SqlClient.SqlVisitor.VisitExpression(SqlExpression exp)
   at System.Data.Linq.SqlClient.SqlVisitor.VisitUnaryOperator(SqlUnary uo)
   at System.Data.Linq.SqlClient.PostBindDotNetConverter.Visitor.VisitUnaryOperator(SqlUnary uo)
```

Also other query options isn't 100$ supported. For example, `$select=Id,Name` works, but $select=Id,Name&$expand=Orders fails with:


```
Could not translate expression 'Table(Customer).Select(Param_0 => new SelectSome`1() {ModelID = \"8bdc9a3a-5ff6-486e-bbe3-53c69ce10dcf\", Container = new SingleExpandedPropertyWithNext1`1() {Name = \"Order\", Value = new SelectAll`1() {ModelID = \"8bdc9a3a-5ff6-486e-bbe3-53c69ce10dcf\", Instance = Param_0.Order, UseInstanceForProperties = True}, Next0 = new NamedProperty`1() {Name = \"Id\", Value = Convert(Param_0.Id)}, Next1 = new NamedProperty`1() {Name = \"Name\", Value = Param_0.Name}, IsNull = (Param_0.Order == null)}})' into SQL and could not treat it as a local expression.
```

@mikepizzo  Is it acceptable to resolve that comment by throwing not supported exception when Linq2SQL is used and creating an issue to eventually support Linq2SQL?

Reply:
@mikepizzo Is it ok for me to block Linq2SQL due to issue mentioned above?

Reply:
Yes; that's fine with me, as long as you throw a clear error for the new scenarios and aren't blocking scenarios that worked previously.